### ISSUE TITLE
on DFS scan violation discovery, return better options

### ISSUE THREAD

**Issue by michbarsinai** [bThink-BGU/BPjs#204]
The DFS violation listener should return one of the following:
* CONTINUE
* PRUNE
* HALT

**Comment by achiyae** [comment]
I want different behaviors (Continue, prone, halt) for different violations. Will it be supported?

**Comment by michbarsinai** [comment]
Yes, your progress listener will get the violation can can then decide what to do.
